### PR TITLE
Evaluate/potentially remove unused code

### LINKED ISSUE #1036: `Result` usages should get rewritten to throw instead
In the future all similar `Result` usages should get rewritten to throw instead, if the goal is to communicate success / failure. 

Besides, current user should always come through method parameters, as per #863 . 

Filing an issue for the `Result` refactor.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/pull/1030#discussion_r752207136_

### LINKED ISSUE #1054: use `current` instead of `actual` to refer to logged in user
should be maybe called `currentUser`? `actual` to me implies real rather than current.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/pull/1030#discussion_r753670374_

### LINKED ISSUE #1080: evaluate / potentially remove unused code
what about unused code? 
```
Item< func performSomeNewSchema [private] from: FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift:150:0>
Item< enum Constants [private] from: FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift:60:0>
Item< var cancellable [private] from: FlowCrypt/Functionality/FilesManager/FilesManager.swift:35:0>
Item< func performGmailApiMigration [private] from: FlowCrypt/Functionality/Migration/DBMigrationService.swift:41:0>
Item< var logger [private lazy] from: FlowCrypt/Functionality/Services/Local Private Key Services/KeyDataStorage.swift:13:0>
Item< var cancellable [private] from: FlowCrypt/Functionality/Services/Backup Services/BackupService.swift:16:0>
Item< enum CodingKeys [private] from: FlowCrypt/Functionality/Services/Account Server Services/EnterpriseServerApi.swift:53:0>
Item< func url [private static] from: FlowCrypt/Functionality/Services/Account Server Services/BackendApi.swift:14:0>
 Item< func normalize [private static] from: FlowCrypt/Functionality/Services/Account Server Services/BackendApi.swift:18:0>
Item< let attester [private] from: FlowCrypt/Controllers/Settings/Backup/Backups Option Scene/BackupOptionsViewController.swift:34:0>
Item< var logger [private lazy] from: FlowCrypt/Controllers/Compose/ComposeViewController.swift:25:0>
 Item< var composedLatestDraft [private] from: FlowCrypt/Controllers/Compose/ComposeViewController.swift:67:0>
Item< var logger [private lazy] from: FlowCrypt/Controllers/Setup/SetupEKMKeyViewController.swift:32:0>
Item< var logger [private lazy] from: FlowCrypt/Controllers/Setup/SetupGenerateKeyViewController.swift:38:0>
Item< func handleAnotherKeySelection [private] from: FlowCrypt/Controllers/Setup/SetupManuallyEnterPassPhraseViewController.swift:291:0>
Item< var logger [private lazy] from: FlowCrypt/Controllers/SignIn/SignInViewController.swift:31:0>
Item< func openDraft [private] from: FlowCrypt/Controllers/MessageList Extension/MsgListViewController.swift:30:0>
```

_Originally posted by @Kharchevskyi in https://github.com/FlowCrypt/flowcrypt-ios/issues/1064#issuecomment-976866076_

### PR DESCRIPTION
**Pull request by ivan-ushakov**
close #1080
close #1036 
close #1054

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by ivan-ushakov** [comment]
Decided to close also some small issues here

File: FlowCrypt/Controllers/SetupImap/SetupImapViewController.swift [hunk 1]
```diff
@@ -486,16 +486,20 @@ extension SetupImapViewController {
...
+        do {
+            try checkCurrentUser()
             checkImapSession()
+        } catch {
+            switch error as? UserError {
+            case .empty:
+                showToast("other_provider_error_other".localized)
+            case .password:
+                showToast("other_provider_error_password".localized)
+            case .email:
```
Comment (ivan-ushakov) [inline FlowCrypt/Controllers/SetupImap/SetupImapViewController.swift:498]: Interesting that we don't throw this error

Comment (tomholub) [reply FlowCrypt/Controllers/SetupImap/SetupImapViewController.swift:498]: Yes, this error handling possibly needs reworking in general. There is no handling of other errors, also.

Comment (tomholub) [reply FlowCrypt/Controllers/SetupImap/SetupImapViewController.swift:498]: But for now never mind, IMAP/SMTP is low priority atm.

**Review by tomholub — APPROVED** [review]

**Pull request merged**